metrics/wer/README.md

```diff
@@ -0,0 +1,118 @@
+
+The general difficulty of measuring performance lies in the fact that the recognized word sequence can have a different length from the reference word sequence (supposedly the correct one). The WER is derived from the [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance), working at the word level instead of the phoneme level. 
+
+This problem is solved by first aligning the recognized word sequence with the reference (spoken) word sequence using dynamic string alignment. Examination of this issue is seen through a theory called the power law that states the correlation between [perplexity](https://huggingface.co/metrics/perplexity) and word error rate.
+
+Word error rate can then be computed as:
+
+`WER = (S + D + I) / N = (S + D + I) / (S + D + C)`
+
+where
+
+`S` is the number of substitutions,
+
+`D` is the number of deletions,
+
+`I` is the number of insertions,
+
+`C` is the number of correct words,
+
+`N` is the number of words in the reference (`N=S+D+C`).
+
+
+## How to use 
+
+The metric takes two inputs: references (a list of references for each speech input) and predictions (a list of transcriptions to score).
+
+
+```python
+from datasets import load_metric
+wer = load_metric("wer")
+wer_score = wer.compute(predictions=predictions, references=references)
+```
+## Output values
+
+This metric outputs a float representing the word error rate.
+
+```
+print(wer_score)
+0.5
+```
+
+This value indicates the average number of errors per reference word. 
+
+The **lower** the value, the **better** the performance of the ASR system, with a WER of 0 being a perfect score.
+
+## Values from popular papers
+
+While initial papers such as [Graves (2014)](http://proceedings.mlr.press/v32/graves14.pdf) reported a WER of 0.27, recent systems such as [Word2vec 2.0](https://arxiv.org/pdf/2006.11477.pdf) have a much lower WER, ranging from 0.018 to 0.033). 
+
+WER values will depend on the dataset being used for evaluation, with datasets such as [LibriSpeech](https://huggingface.co/datasets/librispeech_asr), [Common Voice](https://huggingface.co/datasets/common_voice) and [Timit](https://huggingface.co/datasets/timit_asr) commonly used for evaluating ASR systems.
+
+## Examples 
+
+Perfect match between prediction and reference:
+
+```python
+from datasets import load_metric
+wer = load_metric("wer")
+predictions = ["hello world", "good night moon"]
+references = ["hello world", "good night moon"]
+wer_score = wer.compute(predictions=predictions, references=references)
+print(wer_score)
+0.0
+```
+
+Partial match between prediction and reference:
+
+```python
+from datasets import load_metric
+wer = load_metric("wer")
+predictions = ["this is the prediction", "there is an other sample"]
+references = ["this is the reference", "there is another one"]
+wer_score = wer.compute(predictions=predictions, references=references)
+print(wer_score)
+0.5
+```
```

Comment:
```suggestion
```

The `wer_score` of 0.5 can be computed as follows:

`WER = (S + D + I) / N = (3 + 0 + 1) / 8 = 4/8 = 0.5`, with

- 3 substitions: *prediction* -> *reference*, *an* -> *another*, and *sample* -> *one*
- 1 insertion: *other*
```